File: src/main/java/uk/laxd/dndSimulator/condition/Dead.java
package uk.laxd.dndSimulator.condition;

import uk.laxd.dndSimulator.character.Character;
import uk.laxd.dndSimulator.dice.Die;

public class Dead extends Unconscious {
    private int goodRolls;
    private int badRolls;

    @Override
    public void onTurnStart(Character character) {
        int rollResult = Die.D20.roll();

        if(rollResult == 1) {
            badRolls += 2;
        }
        else if(rollResult > 1 && rollResult < 10) {
            badRolls += 1;
        }
        else if(rollResult >= 10 && rollResult < 20) {
            goodRolls += 2;
        }
        else if(rollResult == 20) {
            character.setHp(1);
            // TODO: Remove this from character effects
        }

        if(badRolls >= 3) {
            // Death!
        }
        else if(goodRolls >= 3) {
            // Stabilise
        }
    }
}

File: app/src/main/kotlin/uk/laxd/dndSimulator/condition/Dead.kt
package uk.laxd.dndSimulator.condition

import uk.laxd.dndSimulator.character.Character
import uk.laxd.dndSimulator.dice.Die

class Dead : Condition("Dead") {

    private var goodRolls = 0
    private var badRolls = 0

    override fun onTurnStart(character: Character) {
        when (Die.D20.roll()) {
            1 -> badRolls += 2
            in 2..9 -> badRolls += 1
            in 10..19 -> goodRolls += 1
            20 -> goodRolls += 2
        }

        if (badRolls >= 3) {
            // Death!
        } else if (goodRolls >= 3) {
            // Stabilise
        }
    }
}